### PR TITLE
Fund all

### PR DESCRIPTION
**Pull request by wythe**
This is a work in progress for #665.

I have the basic feature implemented.  I want to remove redundant code and make testing more comprehensive.  Also, there is a valgrind error in one of the tests.

### CODE REVIEW

File: lightningd/opening_control.c [hunk 1]
```diff
@@ -811,54 +811,79 @@ static void gossip_peer_released(struct subd *gossip,
...
+			BITCOIN_SCRIPTPUBKEY_P2WSH_LEN,
+			&fc->funding_satoshi,
+			&fee_estimate);
+		if (!fc->utxomap || fc->funding_satoshi < 546) {
+			command_fail(cmd, "Cannot afford fee %"PRIu64,
+				     fee_estimate);
+			return;
+		}
+		fc->change = 0;
+    } else {
```
Comment (ZmnSCPxj) [inline lightningd/opening_control.c]: Need to set `fc->change_keyindex = 0` too, to quiet valgrind.

File: cli/lightning-cli.c [hunk 2]
```diff
@@ -271,7 +271,9 @@ int main(int argc, char *argv[])
 			add_input(&cmd, argv[i], i, argc);
 		tal_append_fmt(&cmd, "] }");
 	}
-
+#if 0
```
Comment (ZmnSCPxj) [inline cli/lightning-cli.c]: Unnecessary change.

File: cli/lightning-cli.c [hunk 3]
```diff
@@ -341,7 +343,6 @@ int main(int argc, char *argv[])
 		opt_free_table();
 		return 0;
 	}
-
```
Comment (ZmnSCPxj) [inline cli/lightning-cli.c]: Unnecessary change.

File: lightningd/opening_control.c [hunk 4]
```diff
@@ -811,54 +811,79 @@ static void gossip_peer_released(struct subd *gossip,
...
 
+	fc = tal(cmd, struct funding_channel);
 	fc->cmd = cmd;
 
-	if (!pubkey_from_hexstr(buffer + peertok->start,
-				peertok->end - peertok->start, &fc->peerid)) {
-		command_fail(cmd, "Could not parse id");
-		return;
-	}
+    fc->funding_satoshi = 0;
```
Comment (ZmnSCPxj) [inline lightningd/opening_control.c]: Bad indentation.

File: lightningd/opening_control.c (hunk 1 above)
Comment (ZmnSCPxj) [inline lightningd/opening_control.c]: Indentation please.

File: lightningd/opening_control.c [hunk 5]
```diff
@@ -811,54 +811,79 @@ static void gossip_peer_released(struct subd *gossip,
...
+			&fc->funding_satoshi,
+			&fee_estimate);
+		if (!fc->utxomap || fc->funding_satoshi < 546) {
+			command_fail(cmd, "Cannot afford fee %"PRIu64,
+				     fee_estimate);
+			return;
+		}
+		fc->change = 0;
+    } else {
+        fc->utxomap = build_utxos(fc, cmd->ld, fc->funding_satoshi,
```
Comment (ZmnSCPxj) [inline lightningd/opening_control.c]: Insufficient indentation.

**Comment by cdecker** [comment]
Please don't do merge commits, they tend to mess up the history.

**Comment by cdecker** [comment]
Excellent work, this is starting to look very promising. In its current state I'd squash the PR into a single commit, but if you find it better to organize it into multiple commits feel free to split it according to those lines and `git push -f` it here.

Other than the random naming changes I'm happy with the PR as it currently is.

**Comment by cdecker** [comment]
@wythe no worries, I'll just squash the PR, since the only reasonable separation I can think of is to split out the integration test, but that's also a bit artificial.

**Comment by cdecker** [comment]
Congrats @wythe to your first commit to the c-lightning repository :zap:

**Comment by wythe** [comment]
Great.  Thanks for the help.
